### ISSUE TITLE
Mistake?

### ISSUE THREAD

**Issue by brianhill11** [jackyko1991/vnet-tensorflow#12]
Hi there, 

Thank you very much for posting your implementation. I was going through the code and noticed that you batch normalized x, but did you mean to batch normalize layer_input ? Otherwise the residual addition is just adding x to x... 

This also happens in several other parts of the code by the way. 

https://github.com/jackyko1991/vnet-tensorflow/blob/4c609471ba001e4920673c54ae4a0f0285286c99/VNet.py#L48

Thanks,
Brian

**Comment by DylanHsu** [comment]
Hi Brian,

I'll let Jacky comment on your original question, but if you don't mind a couple questions...
 I'm wondering how you found this issue, and if you noticed any changes in the convergence time or the performance after convergence, when changing those lines to what you think is intended. According to the reasoning presented in Milletari et al (2016), a correct implementation of the residual learning should increase the convergence speed, but our group notices good performance with the architecture despite the fact that this appears to be a legitimate bug.

best regards,
Dylan Hsu
Memorial Sloan-Kettering Cancer Center

**Comment by jackyko1991** [comment]
I think this is a bug when i copying the batch_normalization code...
correct one should be 
```python
layer_input = tf.layers.batch_normalization(layer_input, momentum=0.99, epsilon=0.001,center=True, scale=True,training=is_training)
```
I will update the code asap

**Comment by brianhill11** [comment]
> Hi Brian,
> 
> I'll let Jacky comment on your original question, but if you don't mind a couple questions...
> I'm wondering how you found this issue, and if you noticed any changes in the convergence time or the performance after convergence, when changing those lines to what you think is intended. According to the reasoning presented in Milletari et al (2016), a correct implementation of the residual learning should increase the convergence speed, but our group notices good performance with the architecture despite the fact that this appears to be a legitimate bug.
> 
> best regards,
> Dylan Hsu
> Memorial Sloan-Kettering Cancer Center

Hi Dylan, 

I actually didn’t run the code, so I’m not sure about the performance with/without the change. I was looking to implement a 1D version using Keras so I was digging through the code while writing my own and came across this issue. Hopefully the fix further improves the convergence speed and performance. 

Thanks,
Brian

**Comment by brianhill11** [comment]
Issue looks to have been fixed by @jackyko1991 in https://github.com/jackyko1991/vnet-tensorflow/commit/381fbd1fe97460fab1fdf975b887a931b2ff9967

**Comment by DylanHsu** [comment]
Hi, just want to add my experience is that the convergence speed/stability improves with the updated model. Dylan
